### PR TITLE
MGMT-19348: Mark cluster-create-param high_availability_mode as deprecated

### PR DESCRIPTION
**Pull request by danmanor**
Since new param `control_plane_count` was introduced, `high_availability_mode` is redundant.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -4782,7 +4782,7 @@ definitions:
         enum: ['Full', 'None']
         default: 'Full'
         description: |
```
Comment (omertuc) [inline swagger.yaml:4784]:
```suggestion
        default: 'Full'
        deprecated: true
        description: |
```

File: swagger.yaml [hunk 2]
```diff
@@ -4782,7 +4782,7 @@ definitions:
...
-          Guaranteed availability of the installed cluster. 'Full' installs a Highly-Available cluster
+          (DEPRECATED) Guaranteed availability of the installed cluster. 'Full' installs a Highly-Available cluster
```
Comment (omertuc) [inline swagger.yaml:4785]:
```suggestion
	"Deprecated, please use ... instead. Guaranteed availability of the installed cluster. 'Full' installs a Highly-Available cluster"
```

**Comment by omertuc** [comment]
Even though it's deprecated, I hope we're gonna keep it at least until we move to some future v3 API

File: swagger.yaml (hunk 1 above)
Comment (danmanor) [reply swagger.yaml:4784]: Marking definitions is API 2.0 is not supported yet - https://stackoverflow.com/questions/49870109/how-to-annotate-a-field-as-deprecated-in-openapi-swagger-2-0 (also got an error trying myself)

File: api/vendor/github.com/openshift/assisted-service/models/cluster_create_params.go [hunk 3]
```diff
@@ -52,7 +52,7 @@ type ClusterCreateParams struct {
 	// Installation disks encryption mode and host roles to be applied.
 	DiskEncryption *DiskEncryption `json:"disk_encryption,omitempty" gorm:"embedded;embeddedPrefix:disk_encryption_"`
 
-	// Guaranteed availability of the installed cluster. 'Full' installs a Highly-Available cluster
+	// (DEPRECATED) please use 'control_plane_count' instead. Guaranteed availability of the installed cluster. 'Full' installs a Highly-Available cluster
```
Comment (rccrdpccl) [inline api/vendor/github.com/openshift/assisted-service/models/cluster_create_params.go:55]: nit:
```suggestion
	// (DEPRECATED) Please use 'control_plane_count' instead. Guaranteed availability of the installed cluster. 'Full' installs a Highly-Available cluster
```
for consistency with other deprecation messages

Comment (rccrdpccl) [reply api/vendor/github.com/openshift/assisted-service/models/cluster_create_params.go:55]: Same for all other spots

Comment (danmanor) [reply api/vendor/github.com/openshift/assisted-service/models/cluster_create_params.go:55]: Thanks !

**Pull request merged**